lenses/struct-lens.rkt

```diff
@@ -0,0 +1,31 @@
+#lang racket/base
+
+(provide struct-lens)
+
+(require racket/local
+         syntax/parse/define
+         alexis/util/struct
+         "main.rkt"
+         (for-syntax racket/base
+                     syntax/parse
+                     racket/syntax
+                     ))
+(module+ test
+  (require rackunit fancy-app (only-in lenses lens-transform*)))
+
+(define-simple-macro (struct-lens s:id fld:id)
+  #:with s-fld      (format-id #'s "~a-~a" #'s #'fld #:source #'fld)
+  #:with s-fld-set  (format-id #'s "~a-~a-set" #'s #'fld #:source #'fld)
+  (local [(define-struct-updaters s)]
```

Comment:
In broad strokes, `block` is to `local` as an internal-definition context is to `let`, so either is fine. I personally like `local` here since it semantically expresses the intent, but it's stylistic.